File: 2997-minimum-number-of-operations-to-make-array-xor-equal-to-k/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.java
class Solution {
    public int minOperations(int[] nums, int k) {
        int xor=0;
        for(int i:nums){
            xor^=i;
        }
        int diff = xor^k;
        int res = 0;
        while(diff>0){
            diff = diff&(diff-1);
            res+=1;
        }
        return res;
    }
}

File: 2997-minimum-number-of-operations-to-make-array-xor-equal-to-k/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        xor=0
        for i in nums:
            xor ^= i
        diff=0
        while xor!=0 or k!=0:
            if xor&1 != k&1:
                diff+=1
            xor>>=1
            k>>=1
        return diff